pycode/memilio-simulation/memilio/simulation/bindings/models/abm.cpp

```diff
@@ -134,6 +117,26 @@ PYBIND11_MODULE(_simulation_abm, m)
         .def(py::self += mio::abm::TimeSpan{})
         .def(py::self - mio::abm::TimeSpan{})
         .def(py::self -= mio::abm::TimeSpan{});
+        
+    pymio::bind_class<mio::abm::TestParameters, pymio::EnablePickling::Never>(m, "TestParameters")
+        .def(py::init<double, double, mio::abm::TimeSpan, mio::abm::TestType>())
+        .def_readwrite("sensitivity", &mio::abm::TestParameters::sensitivity)
+        .def_readwrite("specificity", &mio::abm::TestParameters::specificity)
+        .def_readwrite("required_time", &mio::abm::TestParameters::required_time)
+        .def_readwrite("type", &mio::abm::TestParameters::type);
+
+    pymio::bind_CustomIndexArray<mio::UncertainValue<double>, mio::abm::VirusVariant, mio::AgeGroup>(
+        m, "_AgeParameterArray");
+    pymio::bind_CustomIndexArray<mio::abm::TestParameters, mio::abm::TestType>(m, "_TestData");
+    pymio::bind_Index<mio::abm::ExposureType>(m, "ExposureTypeIndex");
+    pymio::bind_ParameterSet<mio::abm::ParametersBase, pymio::EnablePickling::Never>(m, "ParametersBase");
+    pymio::bind_class<mio::abm::Parameters, pymio::EnablePickling::Never, mio::abm::ParametersBase>(m, "Parameters")
+        .def(py::init<int>())
+        .def("check_constraints", &mio::abm::Parameters::check_constraints);
```

Comment:
I wonder why this is marked private/hidden (the leading underscore)? If it is not meant to be accessed by the user, why do we bind it?

Reply:
So the types are needed, e.g. when accessing the TestData Parameter. While the bindings build without issue when commenting out the _TestData bind, the test fails with the following message:

```
ERROR: test_locations (memilio.simulation_test.test_abm.TestAbm)
----------------------------------------------------------------------
TypeError: Unregistered type : mio::CustomIndexArray<mio::abm::TestParameters, mio::abm::TestType>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/schm_r6/Documents/memilio/pycode/memilio-simulation/memilio/simulation_test/test_abm.py", line 60, in test_locations
    1), t0, t0 + abm.days(1), model.parameters.TestData[abm.TestType.Antigen], 1.0)
TypeError: Unable to convert function return value to a Python type! The signature was
        (arg0: memilio.simulation._simulation_abm.ParametersBase) -> mio::CustomIndexArray<mio::abm::TestParameters, mio::abm::TestType>
```